internal/reflect.go

```diff
@@ -31,10 +31,17 @@ func init() {
 	interfaceTyp = reflect.TypeOf(&x).Elem()
 }
 
-// ToArray converts a []interface{} slice into an equivalent fixed-length array
+// MakeHashable converts a []interface{} slice into an equivalent fixed-length array
 // [...]interface{} for use as a comparable map key
-//
-func ToArray(s []interface{}) interface{} {
+func MakeHashable(s []interface{}) interface{} {
```

Comment:
My concern would be breaking things at the SQL layer if I put it higher up. If I were to pick out a refactor that could be worth doing, it's making the filter more responsible for its various conversions so that we don't have to hold in our heads what these things ^ are used for.

Reply:
Looks like batching doesn't actually use the `driver.Value`. Need to find a way to do so.